nixos/modules/installer/tools/nixos-generate-config/default.nix

```diff
@@ -0,0 +1,79 @@
+with import <nixpkgs> {};
```

Comment:
Instead of relying on a channel, shouldn't this become `with import ../../../..` (or whatever the correct level of nesting is)?

Reply:
Ideally this should be some sort of `callPackage` invocation rather than re-importing nixpkgs.
I might add a shell.nix convenience wrapper though for incremental compilation.

Reply:
This should be a "normal" function like every other package; it shouldn't import nixpkgs at all. Also it shouldn't need a separate `shell.nix`, since `nix-shell '<nixpkgs/nixos>' -A config.system.build.nixos-generate-config` should work.

Reply:
Ok. I will document the use of `nix-shell` than in a small README.